src/MigrationTools.Clients.AzureDevops.ObjectModel/Enrichers/TfsEmbededImagesEnricher.cs

```diff
@@ -48,71 +51,99 @@ public override int Enrich(WorkItemData sourceWorkItem, WorkItemData targetWorkI
-                                        {
-                                            result.EnsureSuccessStatusCode();
-                                        }
+                                        result.EnsureSuccessStatusCode();
                                     }
                                 }
+                            }
 
-                                if (GetImageFormat(File.ReadAllBytes(fullImageFilePath)) == ImageFormat.unknown)
-                                {
-                                    throw new Exception($"Downloaded image [{fullImageFilePath}] from Work Item [{wi.ToWorkItem().Id}] Field: [{field.Name}] could not be identified as an image. Authentication issue?");
-                                }
-
-                                int attachmentIndex = wi.ToWorkItem().Attachments.Add(new Attachment(fullImageFilePath));
-                                wi.SaveToAzureDevOps();
+                            if (GetImageFormat(File.ReadAllBytes(fullImageFilePath)) == ImageFormat.unknown)
+                            {
+                                throw new Exception($"{logTypeName}: Downloaded image [{fullImageFilePath}] from Work Item [{wi.ToWorkItem().Id}] Field: [{field.Name}] could not be identified as an image. Authentication issue?");
+                            }
 
-                                var newImageLink = wi.ToWorkItem().Attachments[attachmentIndex].Uri.ToString();
+                            var attachmentInfo = new Microsoft.TeamFoundation.WorkItemTracking.Internals.AttachmentInfo(fullImageFilePath);
+                            wi.ToWorkItem().UploadAttachment(attachmentInfo);
 
+                            if (attachmentInfo.IsUploaded)
+                            {
+                                var newImageLink = BuildAttachmentUrl(attachmentInfo);
                                 field.Value = field.Value.ToString().Replace(match.Value, newImageLink);
-                                wi.ToWorkItem().Attachments.RemoveAt(attachmentIndex);
-                                wi.SaveToAzureDevOps();
-                                File.Delete(fullImageFilePath);
                             }
+                            else
+                            {
+                                throw new Exception($"{logTypeName}: Unable to upload the image [{fullImageFilePath}] to Work Item [{wi.ToWorkItem().Id}] Field: [{field.Name}].");
+                            }
+                        }
+                        finally
+                        {
+                            if (File.Exists(fullImageFilePath))
+                                File.Delete(fullImageFilePath);
                         }
                     }
                 }
+                catch (Exception ex)
+                {
+                    Log.LogError(ex, $"{logTypeName}: Unable to fix HTML field attachments for work item {wi.Id} from {oldTfsurl} to {newTfsurl}");
```

Comment:
Please return all "LogInformation" lines to the format `"test {item} and {item2}", item, itesm2` format. This is a deliberate setup using a logging platform that converts variables to telemetry and highlights. Also for Exceptions.

Reply:
Updated as requested..